### PR TITLE
feat: replace sensitive values and allow removing values

### LINKED ISSUE #27: filter out sensitive fields and store them as required on storage creation
if we get an rclone config with sensitive fields, instead of erroring out, we should remove them from the stored config and track them as required in a separate field

### PR DESCRIPTION
**Pull request by Panaetius**
closes #27

### CODE REVIEW

**Review by olevski — CHANGES_REQUESTED** [review]
I am really confused how setting the value to `<sensitive>` resolves this issue. Shouldn't we be doing more checks when we handle this field? Or does rclone or the ui understand the meaning of this? What happens if someone is "smart" enough to set the field to such a value in a request that is coming from the client?

File: components/renku_data_services/storage_schemas/core.py [hunk 1]
```diff
@@ -155,12 +155,12 @@ def matches_provider(self, provider: str | None) -> bool:
...
             provider_check = provider_check[0].split(",")
 
-        return provider in provider_check == match_type
+        return (provider in provider_check) == match_type
 
     def validate_config(self, value, provider: str | None):
         """Validate an RClone option."""
         if self.sensitive or self.is_password:
-            raise errors.ValidationError(message=f"Field '{self.name}' is sensitive and should not be set.")
+            return "<sensitive>"
```
[the remark this answers is not in the document]
Comment (Panaetius) [reply components/renku_data_services/storage_schemas/core.py:163]: the UI uses the placeholder. It's not rclone specific, just before we removed sensitive values and now we replace them with the placeholders so we know what sensitive values were there

[the remark this answers is not in the document]
Comment (olevski) [reply components/renku_data_services/storage_schemas/core.py:163]: Can you add this in the docstring for the validate_config method (simply that the ui recognizes and depends on the `<sensivitve>` value? I can totally see someone changing the value and not being aware that the UI expects that value.

**Review by olevski — APPROVED** [review]

**Review by olevski — APPROVED** [review]

File: components/renku_data_services/storage_schemas/core.py [hunk 2]
```diff
@@ -155,12 +155,12 @@ def matches_provider(self, provider: str | None) -> bool:
         if "," in provider_check[0]:
             provider_check = provider_check[0].split(",")
 
-        return provider in provider_check == match_type
+        return (provider in provider_check) == match_type
 
     def validate_config(self, value, provider: str | None):
```
[the remark this answers is not in the document]
Comment (Panaetius) [reply components/renku_data_services/storage_schemas/core.py:160]: addressed in https://github.com/SwissDataScienceCenter/renku-data-services/pull/41

**Review by olevski — APPROVED** [review]

**Pull request merged**